File: services/auth/src/main/java/com/example/iter/auth/domain/repository/UserAddressRepository.java
package com.example.iter.auth.domain.repository;

import com.example.iter.auth.domain.entity.UserAddress;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.Optional;

public interface UserAddressRepository extends JpaRepository<UserAddress, Long> {

    Optional<UserAddress> findByUserIdAndDefaultAddressTrue(Long userId);

    long countByUserIdAndDefaultAddressTrue(Long userId);
}

File: services/auth/src/main/kotlin/com/example/iter/auth/domain/repository/UserAddressRepository.kt
package com.example.iter.auth.domain.repository

import com.example.iter.auth.domain.entity.UserAddress
import org.springframework.data.jpa.repository.JpaRepository
import java.util.Optional

interface UserAddressRepository : JpaRepository<UserAddress, Long> {

    fun findByUserIdAndDefaultAddressTrue(userId: Long): Optional<UserAddress>

    fun countByUserIdAndDefaultAddressTrue(userId: Long): Long
}
